src/RequestInformation.cs

```diff
@@ -75,10 +75,42 @@ public class RequestInformation
         /// </summary>
         public IDictionary<string, object> QueryParameters { get; set; } = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
         /// <summary>
+        /// Vanity method to add the query parameters to the request query parameters dictionary.
+        /// </summary>
+        /// <param name="source">The query parameters to add.</param>
+        public void AddQueryParameters(object source)
+        {
+            if(source == null) return;
+            foreach(var property in source.GetType()
+                                        .GetProperties()
+                                        .Select(
+                                            x => (
+                                                Name: x.GetCustomAttributes(false)
+                                                    .OfType<QueryParameterAttribute>()
+                                                    .FirstOrDefault()?.TemplateName ?? x.Name.ToFirstCharacterLowerCase(),
+                                                Value: x.GetValue(source)
+                                            )
+                                        )
+                                        .Where(x => x.Value != null && !QueryParameters.ContainsKey(x.Name)))
+            {
+                QueryParameters.Add(property.Name, property.Value);
+            }
+        }
+        /// <summary>
         /// The Request Headers.
         /// </summary>
         public IDictionary<string, string> Headers { get; set; } = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
         /// <summary>
+        /// Adds request headers to the request.
+        /// </summary>
+        /// <param name="source">The request headers to add.</param>
+        public void AddHeaders(IDictionary<string, string> source)
+        {
+            if(source == null) return;
+            foreach(var header in source)
+                Headers.Add(header.Key, header.Value);
```

Comment:
Do you think we should maybe use `TryAdd`? or do we want the user to possibly experience an exception?
```suggestion
                Headers.TryAdd(header.Key, header.Value);
```

Reply:
added an extension method. thanks for the callout.